stdlib/src/builtin/builtin_slice.mojo

```diff
@@ -158,11 +146,41 @@ struct Slice(Sized, Stringable, EqualityComparable):
         Returns:
             The slice index.
         """
-        return self.start + idx * self.step
+        return self.start.value() + idx * self.step
 
     @always_inline("nodebug")
     fn _has_end(self) -> Bool:
-        return self.end != _int_max_value()
+        return self.end.__bool__()
+
+    fn adjust(inout self, src_len: Int):
```

Comment:
I'm not sure if this should be public. @ConnorGray WDYT, could we just rename it to `_adjust`?

Reply:
I agree, this should probably be private.

It would also be great to see some more docs for this. The current description somewhat assumes I know what it means to "adjust" this slice index. Is this really more like a "normalization" or "canonicalization" kind of operation?

Reply:
I've changed the function name and added more details to the description with some examples.